src/renderer/renderElement.ts

```diff
@@ -1,12 +1,66 @@
-import { ExcalidrawElement } from "../element/types";
+import { ExcalidrawElement, ExcalidrawTextElement } from "../element/types";
 import { isTextElement } from "../element/typeChecks";
-import { getDiamondPoints, getArrowPoints } from "../element/bounds";
+import {
+  getDiamondPoints,
+  getArrowPoints,
+  getElementAbsoluteCoords,
+} from "../element/bounds";
 import { RoughCanvas } from "roughjs/bin/canvas";
 import { Drawable } from "roughjs/bin/core";
 import { Point } from "roughjs/bin/geometry";
 import { RoughSVG } from "roughjs/bin/svg";
 import { RoughGenerator } from "roughjs/bin/generator";
-import { SVG_NS } from "../utils";
+import { SVG_NS, distance } from "../utils";
+import rough from "roughjs/bin/rough";
+
+function generateElementCanvas(element: ExcalidrawElement) {
+  const canvas = document.createElement("canvas");
+  var context = canvas.getContext("2d")!;
+
+  const isLinear = /\b(arrow|line)\b/.test(element.type);
+
+  if (isLinear) {
+    const [x1, y1, x2, y2] = getElementAbsoluteCoords(element);
+    canvas.width = (distance(x1, x2) + 40) * window.devicePixelRatio;
+    canvas.height = (distance(y1, y2) + 40) * window.devicePixelRatio;
+    element.canvasOffsetX =
+      element.x > x1 ? distance(element.x, x1) * window.devicePixelRatio : 0;
```

Comment:
So, the problem with lines is, their individual points are not normalized and go to the negative. This if we drew them to the temp canvas, it would overflow the context. Thus, we need to offset the context by the delta:

![image](https://user-images.githubusercontent.com/5153846/74239866-07299000-4cd9-11ea-9173-823b026c3d9f.png)